src/System.Private.CoreLib/shared/Interop/Unix/System.Globalization.Native/Interop.Utils.cs

```diff
@@ -13,39 +15,44 @@ internal static partial class Interop
     /// increasing buffer until the size is big enough.
     /// </summary>
     internal static bool CallStringMethod<TArg1, TArg2, TArg3>(
-        Func<TArg1, TArg2, TArg3, StringBuilder, Interop.Globalization.ResultCode> interopCall,
+        SpanFunc<char, TArg1, TArg2, TArg3, Interop.Globalization.ResultCode> interopCall,
         TArg1 arg1,
         TArg2 arg2,
         TArg3 arg3,
         out string result)
     {
-        const int initialStringSize = 80;
-        const int maxDoubleAttempts = 5;
+        const int InitialStringSize = 256;
+        const int MaxDoubleAttempts = 3;
 
-        StringBuilder stringBuilder = StringBuilderCache.Acquire(initialStringSize);
+        Span<char> buffer = stackalloc char[InitialStringSize];
 
-        for (int i = 0; i < maxDoubleAttempts; i++)
+        for (int i = 0; i < MaxDoubleAttempts; i++)
```

Comment:
>MaxDoubleAttempts [](start = 28, length = 17)

I don't think we really need this number of trials. if the first one failed then we can allocate char[] with the max and try only one more.

Reply:
Ok.  Will just start at 256 on the stack and jump to 1280 on the heap (I'm not sure where that came from and whether it's actually meaningful, but that's the max that would have previously been used).